src/honey_badger/epoch_state.rs

```diff
@@ -205,15 +209,27 @@ where
             decryption: BTreeMap::default(),
             accepted_proposers: Default::default(),
             subset_handler: subset_handling_strategy.into(),
+            require_decryption,
             _phantom: PhantomData,
         })
     }
 
     /// If the instance hasn't terminated yet, inputs our encrypted contribution.
-    pub fn propose(&mut self, ciphertext: &Ciphertext) -> Result<Step<C, N>> {
-        let ser_ct =
-            bincode::serialize(ciphertext).map_err(|err| ErrorKind::ProposeBincode(*err))?;
-        let cs_step = self.subset.handle_input(ser_ct)?;
+    pub fn propose(
+        &mut self,
+        proposal: Vec<u8>,
```

Comment:
Sorry, I meant to also move the serialization (not only of the ciphertext, but also of the proposal in the first place) into `EpochState`: It is already generic in the type `C` and handles deserialization, and this would make it more consistent. So it should probably be `proposal: &C` here.